### ISSUE TITLE
Accessibility issues coming in MUI components

### ISSUE THREAD

**Issue by huzaifaat** [mui/mui-x#9414]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Link to live example:

Steps:

1.Go to https://mui.com/x/react-data-grid/pagination/.
2. Test the accessibility of website using WAVE extension.
3.In Pagination, after rows per page, it shows missing form label error on dropdown icon. Same behavior in case of MUI Select where dropdown icon displays. Same is the behavior in my application.

### Current behavior 😯

Showing form label missing error after rows per page on the dropdown icon, as well as in MUI Select component where dropdown icon shows.

### Expected behavior 🤔

No error should be displayed in terms of accessibility.

### Context 🔦

I am developing a react application using MUI (Core & X) & having issues in WCAG Accessibility. I am using many of their components but having issues in data grid, select and MUI tabs. I am using the following MUI versions:

"@mui/material": "^5.13.5", "@mui/styled-engine-sc": "^5.8.0", "@mui/x-data-grid": "^5.17.8",

Issues found are:

Wave Evaluation Tool (i.e Accessibility test) is giving Form Label Missing Error wherever there's a dropdown arrow either in MUI Data Grid(after rows per page)
Also in any of MUI Select Component (dropdown for selection) . https://justpaste.it/4q585 [[Data Grid](https://i.stack.imgur.com/6GzNy.png)](https://i.stack.imgur.com/3U3Cv.png)

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>
